frontend/src/components/CheckboxGroup/CheckboxGroupValidator.tsx

```diff
@@ -0,0 +1,221 @@
+import { List } from 'immutable';
+import React, { FunctionComponent, useState } from 'react';
+import { Filters, OperationStepMap } from '../../types/operations';
+import { BasicModal } from '../BasicModal';
+
+interface ComponentProps {
+  steps?: List<OperationStepMap>;
+}
+
+const CheckboxGroupValidator: FunctionComponent<ComponentProps> = (props) => {
+  const [showModal, setShowModal] = useState<boolean>(false);
+  const [modalMessage, setModalMessage] = useState<string>('');
+
+  const validateCheckboxOptions = (
+    checkboxValue: string,
+    checkboxState: boolean | undefined,
+  ): void => {
+    const { steps } = props;
+    steps &&
+      steps.map((step) => {
+        const options = step.get('query_kwargs') as string;
+        toggleModalState(options, checkboxState, checkboxValue);
+      });
+  };
+
+  const toggleModalState = (
+    options: string,
+    checkboxState: boolean | undefined,
+    checkboxValue: string,
+  ): void => {
+    const parsedOptions = parseOptions(options);
```

Comment:
I'm sure this `parseOptions` function will go into utils as well... Should leave us with a much cleaner component.

Reply:
Updated